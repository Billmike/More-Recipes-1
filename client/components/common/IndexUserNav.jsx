import React from 'react';
import PropTypes from 'prop-types';

/**
 * User navigation when unathenticated
 *
 * @param {any} props - The properties passed to the component
 * @returns {object} React element
 */
const IndexUserNav = (props) => {
  const { handleOpenSignin, handleOpenSignup } = props;
  return (
    <ul id="user-control" className="dropdown-content">
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleOpenSignin}
        >
          Sign in
          <i className="material-icons left">person</i>
        </a>
      </li>
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleOpenSignup}
        >
          Register
          <i className="material-icons left">folder</i>
        </a>
      </li>
    </ul>
  );
};

IndexUserNav.propTypes = {
  handleOpenSignup: PropTypes.func.isRequired,
  handleOpenSignin: PropTypes.func.isRequired,
};

export default IndexUserNav;
