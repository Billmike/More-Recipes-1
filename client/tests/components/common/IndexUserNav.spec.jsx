import React from 'react';

import IndexUserNav from '../../../components/common/IndexUserNav';

describe('Index user navigation component', () => {
  const props = {
    handleToggleSigninModal: jest.fn(),
    handleToggleSignupModal: jest.fn(),
  };
  it('should render the component', () => {
    const wrapper = shallow(<IndexUserNav {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate search events', () => {
    const wrapper = shallow(<IndexUserNav {...props} />);

    wrapper.find('#signin').simulate('click');
    expect(props.handleToggleSigninModal).toBeCalled();
    wrapper.find('#register').simulate('click');
    expect(props.handleToggleSignupModal).toBeCalled();
  });
});
