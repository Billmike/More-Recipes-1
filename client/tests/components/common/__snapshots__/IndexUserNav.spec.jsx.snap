// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index user navigation component should render the component 1`] = `
<ul
  className="dropdown-content"
  id="user-control"
>
  <li>
    <a
      className="black-text"
      id="signin"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Sign in
      <i
        className="material-icons left"
      >
        person
      </i>
    </a>
  </li>
  <li>
    <a
      className="black-text"
      id="register"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Register
      <i
        className="material-icons left"
      >
        folder
      </i>
    </a>
  </li>
</ul>
`;
